src/dune_rules/artifact_substitution.ml

```diff
@@ -539,3 +539,69 @@ let copy_file ~conf ?chmod ~src ~dst () =
       Io.close_both (ic, oc);
       Fiber.return ())
     (fun () -> copy ~conf ~input_file:src ~input:(input ic) ~output:(output oc))
+
+let test ~input =
```

Comment:
This function is too similar to `copy`.  The common code should be factored out.